unkeyed/unkey
# feat: Reroll keys

## What does this PR do?
Implements a new feature to Reroll keys.

Fixes #2050 

## Type of change

- [] This change requires a documentation update

## How should this be tested?

- Create a new api
- Create a new key
- View key details
- Click on the button `Reroll Key` and a Dialog will show up
- Select an optional delay period (default is 1h as agreed with @chronark)
- Click on the `Reroll Key` action button
- Wait for the process to finish
- The user must be redirect to the new key details page. 
- Check if the key value has changed.

## Checklist



### Required


### Appreciated


Here a [link](https://www.loom.com/share/cb432dcb8e7b450fa26c7160fcd29530?sid=f1a60be7-5451-49d0-aa3b-21dbf5f35f93) of the new feature recording

#### Reroll Key Dialog
![image](https://github.com/user-attachments/assets/d14ad96f-9615-4b4d-b12e-600a9b573923)

#### Reroll Key Dialog - delay period options
![image](https://github.com/user-attachments/assets/bac8d7d8-b462-4d7c-9a4b-da935258db17)
